src/webextension/list/popup/index.js

```diff
@@ -26,7 +26,7 @@ browser.tabs.query({ active: true, currentWindow: true }).then((result) => {
   const validProtocols = ["http:", "https:"];
   const url = new URL(result[0].url);
   if (validProtocols.includes(url.protocol)) {
-    store.dispatch(filterItems(url.hostname));
+    store.dispatch(filterItems(url.protocol + "//" + url.host));
```

Comment:
I would have expected `url` to be passed in directly here, rather than breaking it up.  This feels like the constraints are spread around the codebase; that will make it harder for us to understand the implications and limitations.

Reply:
We can't (well, we *shouldn't*) pass the URL object, since that goes right into the Redux state, and Redux is supposed to be just JSON. We could make a new Redux action creator, but it'd do pretty much exactly what this does before getting stored in Redux.

Reply:
That is a reasonable argument against passing the URL directly.

However, this still needs to be `hostname` and not `host`.  It needs to keep the optional port, or it risks false positives.  For the vast majority of people for the vast majority of their entries, `hostname` and `host` are identical, so I think false negatives by real users will be rare.

I can see an argument for treating the lack of an explicit port as matching any port, but I'd like to get sign off from security-review.